packages/back-end/src/api/sdk-connections/validations.ts

```diff
@@ -7,17 +7,34 @@ import { getEnvironments } from "../../services/organizations";
 const capabilityParams = [
   ["encryption", "encryptPayload"],
   ["remoteEval", "remoteEvalEnabled"],
+  ["visualEditor", "includeVisualExperiments"],
+  ["redirects", "includeRedirectExperiments"],
 ] as const;
 
 type CababilitiesParamKey = typeof capabilityParams[number][1];
 type CababilitiesParams = { [k in CababilitiesParamKey]?: boolean };
 
 const premiumFeatures = [
   ["encrypt-features-endpoint", "encryptPayload"],
+  ["visual-editor", "includeVisualExperiments"],
+  ["hash-secure-attributes", "hashSecureAttributes"],
+  ["remote-evaluation", "remoteEvalEnabled"],
+  ["redirects", "includeRedirectExperiments"],
+  ["cloud-proxy", "proxyEnabled"],
 ] as const;
 
-type PremiumFeatureKey = typeof premiumFeatures[number][1];
-type PremiumFeatures = { [k in PremiumFeatureKey]?: boolean };
+type PremiumFeatureName = typeof premiumFeatures[number][0];
+type PremiumFeatureParam = typeof premiumFeatures[number][1];
+type PremiumFeatures = { [k in PremiumFeatureParam]?: boolean };
+
+const premiumExtraChecks: {
+  [k in PremiumFeatureName]?: { check: boolean; reason: string };
+} = {
+  "cloud-proxy": {
+    check: !!process.env.IS_CLOUD,
```

Comment:
Import this from `secrets.ts` instead of access `process.env` directly.  We do parsing of raw environment vars and some additional checks, so `IS_CLOUD` will always be a boolean.